persistence/schema/gov.go

```diff
@@ -1,442 +1,145 @@
-		params.AppUnstakingBlocks,
-		params.AppMinimumPauseBlocks,
-		params.AppMaxPauseBlocks,
-
-		params.ServiceNodeMinimumStake,
-		params.ServiceNodeMaxChains,
-		params.ServiceNodeUnstakingBlocks,
-		params.ServiceNodeMinimumPauseBlocks,
-		params.ServiceNodeMaxPauseBlocks,
-		params.ServiceNodesPerSession,
-
-		params.FishermanMinimumStake,
-		params.FishermanMaxChains,
-		params.FishermanUnstakingBlocks,
-		params.FishermanMinimumPauseBlocks,
-		params.FishermanMaxPauseBlocks,
-
-		params.ValidatorMinimumStake,
-		params.ValidatorUnstakingBlocks,
-		params.ValidatorMinimumPauseBlocks,
-		params.ValidatorMaxPauseBlocks,
-		params.ValidatorMaximumMissedBlocks,
+	val := reflect.ValueOf(params)
+	var subQuery string
+	for _, k := range govParamMetadataKeys {
+		pnt := govParamMetadataMap[k]
+		pVal := val.Elem().FieldByName(pnt.PropertyName)
+
+		subQuery += `(`
+		switch govParamMetadataMap[k].PoktValType {
+		case ValTypeString:
+			subQuery += fmt.Sprintf("'%s', %d, true, '%s', '%s'", k, DefaultBigInt, pnt.PoktValType, pVal.Interface())
+		case ValTypeSmallInt, ValTypeBigInt:
+			subQuery += fmt.Sprintf("'%s', %d, true, '%s', %d", k, DefaultBigInt, pnt.PoktValType, pVal.Interface())
+		}
+		subQuery += `),`
+	}
+	return fmt.Sprintf(`INSERT INTO %s VALUES %s`, ParamsTableName, subQuery[:len(subQuery)-1])
+}
 
-		params.ValidatorMaxEvidenceAgeInBlocks,
-		params.ProposerPercentageOfFees,
-		params.MissedBlocksBurnPercentage,
-		params.DoubleSignBurnPercentage,
+func GetParamQuery(paramName string) string {
+	//TODO (@deblasis): clarify if and how we should use `enabled` here
+	return fmt.Sprintf(`SELECT value FROM %s WHERE name=%s AND height=%d and enabled=true`, ParamsTableName, paramName, DefaultBigInt)
+}
 
-		params.MessageDoubleSignFee,
-		params.MessageSendFee,
-		params.MessageStakeFishermanFee,
-		params.MessageEditStakeFishermanFee,
-		params.MessageUnstakeFishermanFee,
-		params.MessagePauseFishermanFee,
-		params.MessageUnpauseFishermanFee,
-		params.MessageFishermanPauseServiceNodeFee,
-		params.MessageTestScoreFee,
-		params.MessageProveTestScoreFee,
+func NullifyParamQuery(paramName string, height int64) string {
+	//TODO (@deblasis): clarify if and how we should use `enabled` here
+	return fmt.Sprintf(`UPDATE %s SET height=%d WHERE name='%s' AND height=%d`, ParamsTableName, height, paramName, DefaultBigInt)
+}
 
-		params.MessageStakeAppFee,
-		params.MessageEditStakeAppFee,
-		params.MessageUnstakeAppFee,
-		params.MessagePauseAppFee,
-		params.MessageUnpauseAppFee,
-		params.MessageStakeValidatorFee,
-		params.MessageEditStakeValidatorFee,
-		params.MessageUnstakeValidatorFee,
-		params.MessagePauseValidatorFee,
-		params.MessageUnpauseValidatorFee,
-		params.MessageStakeServiceNodeFee,
-		params.MessageEditStakeServiceNodeFee,
-		params.MessageUnstakeServiceNodeFee,
-		params.MessagePauseServiceNodeFee,
-		params.MessageUnpauseServiceNodeFee,
-		params.MessageChangeParameterFee,
+type ParamTypes interface {
```

Comment:
s/ParamTypes/ProtobufAttributeTypes/>

Reply:
`ParamTypes` represents the types that the generic can accept:
```go
type ParamTypes interface {
	int | int32 | int64 | []byte | string
}
```
technically it has nothing to do with the protobuf.
Agreed that `ParamTypes` sounds weird though

Reply:
`SupportedParamTypes` or `GovParamTypes`?